app/controllers/users_controller.rb

```diff
@@ -40,8 +41,22 @@ def update
     redirect_to '/profile'
   end
 
+  def update_password
+    current_user.attributes = user_params_2
+    if current_user.save
+      flash[:succes] = "Your information has been updated."
+    else
+      flash[:error] = current_user.errors.full_messages.uniq.to_sentence
+    end
+    redirect_to '/profile'
+  end
+
   private
     def user_params
+      params.require(:user).permit(:name, :address, :city, :state, :zip, :email, :password, :password_confirmation)
+    end
+
+    def user_params_2
       params.permit(:name, :address, :city, :state, :zip, :email, :password, :password_confirmation)
     end
 end
```

Comment:
This seems wrong, but only some things will need to use user_params and others (like password updating) will have to use user_params_2